src/modules/flow/flower-power/flower-power.c

```diff
@@ -0,0 +1,709 @@
+    }
+
+    sol_json_scanner_init_from_token(&scanner, &value);
+    SOL_JSON_SCANNER_OBJECT_LOOP (&scanner, &token, &key, &value, reason) {
+        if (sol_json_token_str_eq(&key, "current_value", strlen("current_value"))) {
+            if (sol_json_token_get_double(&value, &measure->val)) {
+                SOL_WRN("Failed to get current value");
+                return false;
+            }
+            found_cur = true;
+        } else if (sol_json_token_str_eq(&key, "max_threshold",
+                    strlen("max_threshold"))) {
+            if (sol_json_token_get_double(&value, &measure->max))
+                SOL_DBG("Failed to get max value");
+        } else if (sol_json_token_str_eq(&key, "min_threshold",
+                    strlen("min_threshold"))) {
+            if (sol_json_token_get_double(&value, &measure->min))
+                SOL_DBG("Failed to get min value");
+        }
+    }
+
+    if (reason != SOL_JSON_LOOP_REASON_OK)
+        return false;
+
+    return found_cur;
+}
+
+static void
+http_get_cb(void *data, struct sol_http_response *response)
+{
+    struct http_get_data *mdata = data;
+    struct sol_json_scanner scanner, locations_scanner;
+    struct sol_json_token token, key, value;
+    int r;
+    enum sol_json_loop_reason reason;
+    bool found_locations = false;
+
+    RESPONSE_CHECK_API(response, mdata);
+
+    if (response->response_code != 200) {
+        sol_flow_send_error_packet(mdata->node, EINVAL,
+            "Service returned unexpected response code: %d for request %s",
+            response->response_code, STATUS_URL);
+        return;
+    }
+
+    if (!response->content.used) {
+        sol_flow_send_error_packet(mdata->node, EINVAL,
+            "Empty response from server for request %s", STATUS_URL);
+        return;
+    }
+
+    sol_json_scanner_init(&scanner, response->content.data,
+        response->content.used);
+
+    SOL_JSON_SCANNER_OBJECT_LOOP (&scanner, &token, &key, &value, reason) {
+        if (sol_json_token_str_eq(&key, "locations", strlen("locations"))) {
+            found_locations = true;
+            break;
+        }
+    }
+    if (reason != SOL_JSON_LOOP_REASON_OK)
+        goto error;
+
+    if (!found_locations) {
+        SOL_DBG("No plants found on response");
+        return;
+    }
+
+    sol_json_scanner_init_from_token(&locations_scanner, &value);
+    SOL_JSON_SCANNER_ARRAY_LOOP (&locations_scanner, &token,
+        SOL_JSON_TYPE_OBJECT_START, reason) {
+        struct sol_drange fertilizer = {0, -DBL_MAX, DBL_MAX, DBL_MIN};
+        struct sol_drange water = {0, -DBL_MAX, DBL_MAX, DBL_MIN};
+        struct sol_drange temperature = {0, -DBL_MAX, DBL_MAX, DBL_MIN};
+        struct sol_drange light = {0, -DBL_MAX, DBL_MAX, DBL_MIN};
+        char *id = NULL, *timestamp = NULL;
+
+        SOL_JSON_SCANNER_OBJECT_LOOP_NEST (&locations_scanner, &token,
+            &key, &value, reason) {
+            if (sol_json_token_str_eq(&key, "fertilizer",
```

Comment:
I believe there's a macro in `sol-json.h` that helps these comparisons with constant strings (specify them only once, etc).